molviewspec/molviewspec/nodes.py

```diff
@@ -438,6 +438,29 @@ class RepresentationParams(BaseModel):
     type: RepresentationTypeT = Field(description="Representation type, i.e. cartoon, ball-and-stick, etc.")
 
 
+class CartoonParams(RepresentationParams):
+    size_factor: Optional[float] = Field(description="Scales the corresponding visuals.")
+    tubular_helices: Optional[bool] = Field(description="Simplify corkscrew helices to tubes.")
+    # TODO support for variable size, e.g. based on b-factors?
+
+
+class BallAndStickParams(RepresentationParams):
+    ignore_hydrogens: Optional[bool] = Field(descripton="Controls whether hydrogen atoms are drawn.")
+    size_factor: Optional[float] = Field(description="Scales the corresponding visuals.")
+
+
+class SurfaceParams(RepresentationParams):
+    ignore_hydrogens: Optional[bool] = Field(descripton="Controls whether hydrogen atoms are drawn.")
+    size_factor: Optional[float] = Field(description="Scales the corresponding visuals.")
+
+
+RepresentationTypeParams: Dict[RepresentationTypeT, Type[Union[CartoonParams, BallAndStickParams, SurfaceParams]]] = {
+    "cartoon": CartoonParams,
+    "ball_and_stick": BallAndStickParams,
+    "surface": SurfaceParams,
+}
```

Comment:
Like the dictionary approach. Maybe can do this too so the `type` doesn't have to be repeated:

```suggestion
RepresentationTypeParams = {
  t.__fields__["type"].default: t for t in (CartoonParams, BallAndStickParams, SurfaceParams)
}
```

Reply:
Nice, moved to that impl with https://github.com/molstar/mol-view-spec/pull/41/commits/9a46576d468cf166a921cb92437c55eed2817281.